server/data/data-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryStoreImpl.java

```diff
@@ -59,7 +52,7 @@
     private final MailRepositoryStoreConfiguration configuration;
 
     @Inject
-    public MemoryMailRepositoryStore(MailRepositoryUrlStore urlStore, Set<MailRepositoryProvider> mailRepositories, MailRepositoryStoreConfiguration configuration) {
+    public MailRepositoryStoreImpl(MailRepositoryUrlStore urlStore, Set<MailRepositoryProvider> mailRepositories, MailRepositoryStoreConfiguration configuration) {
```

Comment:
Why did you rename that class?

Reply:
Because nothing is related to memory, everything is injected

Reply:
Agree the MailRepoUrlStore API did remove the memory part of this: I back this change.

Reply:
```
    private final MailRepositoryUrlStore urlStore;
    private final Set<MailRepositoryProvider> mailRepositories;
    private final ConcurrentMap<MailRepositoryUrl, MailRepository> destinationToRepositoryAssociations;
    private final Map<Protocol, MailRepositoryProvider> protocolToRepositoryProvider;
    private final Map<Protocol, HierarchicalConfiguration> perProtocolMailRepositoryDefaultConfiguration;
    private final MailRepositoryStoreConfiguration configuration;
```
It looks a lot like memory datastructure

Reply:
It's just the storage of the configuration.
It's the only implementation of MailRepositoryStore which is used for every backend.
I see no reason to keep it in the data-memory project, and by doing so making all the project using it dependend of this project.

Reply:
There are two things I don't agree with:
1. whatever the location of this class, SomethingImpl is always worse than MemorySomething because it doesn't make obvious this is a memory-structure-backed implementation. 
2. regarding the location, putting a service implementation into a -api project doesn't make sense

The problem you probably want to fix is "every backend depends on data-memory". By definition, the data-memory project has very few dependencies because it's written with memory data structures, so I doubt this dependency is really a problem. If you want to minimize that amount of imported classes then you can just split data-api into smaller api projects.

Reply:
Or we rename MailRepositoryStoreImpl to MailRepositoryStore and drop the interface as there is only one implementation and there is no reason to have several ones ?